toolkit/scripts/toolchain/create_toolchain_in_container.sh

```diff
@@ -8,28 +8,38 @@ set -e
 MARINER_BUILD_DIR=$1
 MARINER_SPECS_DIR=$2
 MARINER_SOURCE_URL=$3
+INCREMENTAL_TOOLCHAIN=$4
+ARCHIVE_TOOL=$5
```

Comment:
I'd double-check that archive tool parameter with Andrew. I recall he's been removing these from some scripts a few months ago, but now I don't remember why and if it was here or elsewhere.

Reply:
Looks like https://github.com/microsoft/CBL-Mariner/pull/4026

In this case I think we should keep it.. We know the format of the archive, and this saves several minutes during the build process. (by default `tar` uses the single threaded version of `gzip` which is SUPER slow for this use case.

Reply:
@anphel31 Thoughts?

Reply:
Here's the related PR for main: #3683 
I removed the compression on archives that contained only rpm or srpm files. I left the compression enabled on toolchain_from_container.tar.gz (since it contains individual toolchain files, not compressed rpms). So using the archive tool here should be fine.